src/config.ts

```diff
@@ -41,7 +42,7 @@ try {
 // if envConfigFile set try to get configs
 if (envConfigFile) {
   try {
-    ENV_CONFIG = Object.assign(ENV_CONFIG, configContext(envConfigFile)) as GlobalConfig;
+    merge(configContext(envConfigFile) as GlobalConfig);
   } catch (e) {
     console.warn('Cannot find file ' + envConfigFile.substring(2, envConfigFile.length), 'Using default environment.');
```

Comment:
If this happens (e.g. when you're in dev mode, but don't have a custom environment.dev.js) , buildBaseUrls() is never executed, and the methods that rely on baseUrls break.

Reply:
Good catch. I will fix this. I meant to separate the merge and configContext from the try, but didn't catch the base url construction. Thanks.